tree/ntupleutil/v7/inc/ROOT/RNTupleExporter.hxx

```diff
@@ -0,0 +1,54 @@
+/// \file ROOT/RNTupleExporter.hxx
+/// \ingroup NTuple ROOT7
+/// \author Giacomo Parolini <giacomo.parolini@cern.ch>
+/// \date 2024-12-10
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RNTupleExporter
+#define ROOT7_RNTupleExporter
+
+#include <filesystem>
+#include <vector>
+
+namespace ROOT::Experimental::Internal {
+
+class RPageSource;
+
+struct RExportPagesOptions {
+   enum RExportPageFlags {
+      kNone = 0x0,
+      kIncludeChecksums = 0x1,   
+      kReportExportedFileNames = 0x2,
+
+      kDefaults = kReportExportedFileNames
+   };
+
+   std::filesystem::path fOutputPath = ".";
+   std::uint64_t fFlags = kDefaults;
+};
+
+struct RExportPagesResult {
+   int fNPagesExported;
+   // This only gets filled if kReportExportedFileNames was part of the options flags
+   std::vector<std::string> fExportedFileNames;
```

Comment:
Does this really cost so much that we need it optional? Compared to reading and writing all pages, I would expect filling a `std::vector<std::string>` to be quite cheap...

Reply:
The solution of pre-counting the pages indeed solves the performance issues. There is still a matter of memory being potentially wasted, but it's in the order of 5 MiB for 100k pages. I will remove the flag for now to keep the code simpler; if this ever becomes a concern it's trivial to fix.